src/java/org/apache/cassandra/tools/nodetool/CommandUtils.java

```diff
@@ -0,0 +1,259 @@
+     * subset of tables. This method extracts the table names from the provided cli arguments, assuming
+     * the first argument is the keyspace name and any subsequent arguments are table names.
+     * <p>
+     * If no table names are provided (e.g. only the keyspace is specified), this method returns
+     * an empty array, which signals to the MBeans that the operation should apply to all tables
+     * in the keyspace. This approach provides flexibility to target either all tables or specific
+     * tables as needed.
+     *
+     * @param cmdArgs the list of command arguments, where the first argument is typically the
+     *                keyspace name (ignored), and any subsequent arguments are table names
+     * @return an array of table names, or an empty array (meaning 'all tables') if no extra args are specified.
+     */
+    public static String[] parseOptionalTables(List<String> cmdArgs)
+    {
+        return cmdArgs.size() <= 1 ? EMPTY_STRING_ARRAY : toArray(cmdArgs.subList(1, cmdArgs.size()), String.class);
+    }
+
+    /**
+     * Parses the optional partition key values from the command arguments for nodetool commands.
+     * <p>
+     * Some nodetool commands can operate on a specific partition within a table, which requires
+     * specifying the partition key values after the keyspace and table names in the cli arguments.
+     * This method extracts the partition keys from the provided command arguments, assuming
+     * the first argument is the keyspace name, the second is the table name, and any subsequent
+     * arguments are partition key values.
+     * <p>
+     * If no partition key values are provided (e.g. only keyspace and table are specified), this
+     * method returns an empty array, which signals to the MBeans or command logic that the operation
+     * should apply to all partitions.
+     *
+     * @param cmdArgs the list of command arguments, where the first argument is the keyspace name,
+     *                the second is the table name, and any subsequent arguments are partition key values
+     * @return an array of partition key values, or an empty array if none are specified (meaning 'all partitions').
+     */
+    public static String[] parsePartitionKeys(List<String> cmdArgs)
+    {
+        return cmdArgs.size() <= 2 ? EMPTY_STRING_ARRAY : toArray(cmdArgs.subList(2, cmdArgs.size()), String.class);
+    }
+
+    public static SortedMap<String, SetHostStatWithPort> getOwnershipByDcWithPort(NodeProbe probe, boolean resolveIp,
+                                                                                  Map<String, String> tokenToEndpoint,
+                                                                                  Map<String, Float> ownerships)
+    {
+        SortedMap<String, SetHostStatWithPort> ownershipByDc = Maps.newTreeMap();
+        EndpointSnitchInfoMBean epSnitchInfo = probe.getEndpointSnitchInfoProxy();
```

Comment:
This MBean has been deprecated, so it might be a good idea to use `LocationInfoMBean` instead?

Reply:
True, this needs to be changed. However, I've moved this method "as is" to the utility class so that we don't need to worry about the command behavior changing. I would create a new issue to remove MBean an refactor the places where it's used.
 
https://github.com/apache/cassandra/blob/trunk/src/java/org/apache/cassandra/tools/NodeTool.java#L549C9-L549C83